File: app/src/main/java/fr/bowser/behaviortracker/utils/TimeConverter.java
package fr.bowser.behaviortracker.utils;

public class TimeConverter {

    public static String convertSecondsToHumanTime(long time){

        String string = "";

        long hours = time / 3600;
        if(hours < 10){
            string = "0";
        }
        string = string + hours + ":";

        time %= 3600;

        long minutes = time / 60;
        if(minutes < 10){
            string += "0";
        }
        string = string + minutes + ":";

        time %= 60;

        long seconds = time;
        if(seconds < 10){
            string += "0";
        }
        string += seconds;

        return string;
    }
}

File: app/src/main/java/fr/bowser/behaviortracker/utils/TimeConverter.kt
package fr.bowser.behaviortracker.utils

object TimeConverter {

    @JvmStatic
    fun convertSecondsToHumanTime(time: Long, displayHours: Boolean = true): String {
        if(time < 0){
            throw IllegalStateException("time can't be negative")
        }

        var remainingTime = time
        var string = ""

        if(displayHours) {
            val hours = remainingTime / 3600

            if (hours < 10) {
                string = "0"
            }
            string = "$string$hours:"

            remainingTime %= 3600
        }

        val minutes = remainingTime / 60
        if (minutes < 10) {
            string += "0"
        }
        string = "$string$minutes:"

        remainingTime %= 60

        val seconds = remainingTime
        if (seconds < 10) {
            string += "0"
        }
        string += seconds

        return string
    }

}